js/script.js

```diff
@@ -0,0 +1,213 @@
+        document.getElementById('result').classList.remove('hide');
+        document.getElementById('limit').classList.remove('hide');
+        output.innerHTML = "";
+        result.innerHTML = "";
+    });
+    for (var i = 0; i < buttons.length; i++) { 
+        buttons[i].addEventListener("click", function () {
+            state.userChoice = this.getAttribute("data-move"); 
+            startSingleRound(state.userChoice);
+        });
+    }
+};
+
+function disableButtons(state) {
+    rock.disabled = state;
+    paper.disabled = state;   
+    scissors.disabled = state;    
+};
+
+function roundsLimit() {
+     
+    state.rounds = parseInt(window.prompt('How many round would you play?'));
+    if (state.rounds > 0) {
+       limit.innerHTML = 'We will play up to ' + state.rounds + ' wins';
+    }
+    else if (isNaN(state.rounds) || state.rounds <= 0) {
+       limit.innerHTML = 'Please type correct number of rounds';
+    }
+    return state.rounds ;
+};
+
+function computerChoice() {
+    var compChoice = Math.floor(Math.random() * 3 + 1);
+    if (compChoice === 1) {
+        compChoice = 'rock';
+    } 
+    else if (compChoice === 2) {
+        compChoice = 'paper';    
+    } 
+    else {
+        compChoice = 'scissors';
+    }
+    state.computerChoice = compChoice;
+    return compChoice;
+};
+
+
+
+function compare(userChoice, compChoice) {
+    if (userChoice === compChoice) {
+        state.roundWinner = 'NO ONE WINS this round:';
+        state.lastWinner = "it \'s tie";
+    }
+    else if ((userChoice === "rock" && compChoice === "scissors") || 
+        (userChoice === "paper" && compChoice === "rock")||
+        (userChoice === "scissors" && compChoice === "paper")) {
+        state.roundWinner = 'YOU WON:';
+        state.userScore++;
+        state.lastWinner = "player";          
+    } 
+    else {
+        state.roundWinner = 'COMPUTER WON this round:';
+        state.computerScore++;
+        state.lastWinner = "computer";
+        
+    }
+    return state.roundWinner ;   
+};
+
+
+function clearBox(idElement) {
+    idElement.innerHTML= "";
+};
+
+function roundResultDisplay(roundWinner) {
+    output.insertAdjacentHTML('afterbegin', roundWinner);
+    
+};
+
+function setScore() {
+    document.getElementById('result').innerHTML = (state.userScore + ' - ' + state.computerScore);   
+};
+
+
+function gameIsOver() {
+    if (state.userScore === state.rounds) {  
+        showHideModalContent('YOU WON THE ENTIRE GAME!!!');     
+           
+    }
+    else if (state.computerScore === state.rounds) {
+        showHideModalContent('YOU LOSE, COMPUTER WAS BETTER!!!');        
+    }
+    function hideModal(event) {
+        event.preventDefault();
+        modalOverlay.classList.remove('show');
+    };      
+    for (var i = 0; i < closeButtons.length; i++) {
+        closeButtons[i].addEventListener('click', hideModal);      
+    }
+    modalOverlay.addEventListener('click', hideModal);
```

Comment:
gratuluje :P Twój pierwszy w życiu memory leak :D

dodaj sobie w `hideModal` console loga i zobacz ile razy ta funkcja się odpala w momencie kiedy oczekiwałabyś, że odpali się raz. I sytuacja będzie tym gorsza nim dłużej będziemy grać w grę bez odświeżania strony.

za często rejestrujesz ten i te niżej listenery, pomyśl nad sensownym miejscem gdzie można by je przenieść. Zobacz, że obecnie siedzą w funkcji `gameIsOver`, której celem jest sprawdzić czy gra się skończyła... a nie rejestrować listenery na modale - to wymieszanie odpowiedzialności